src/linalg.jl

```diff
@@ -217,3 +218,32 @@ for FUNC in [:randn!, :rand!]
         return m
     end
 end
+
+
+########### Hadamard Product
+sparse_ranking(a::AbstractMatrix) = 6
+sparse_ranking(a::SDMatrix) = 5
+sparse_ranking(a::SDSparseMatrixCSC) = 4
+sparse_ranking(a::SDPermMatrix) = 3
+sparse_ranking(a::SDDiagonal) = 2
+sparse_ranking(a::IMatrix) = 1
+
```

Comment:
docs?

Reply:
it is not exported.

Reply:
Merge it first, we can add docs later, new feature in YaoBlocks requires hadamard test.